targets/javascript/templates/PlayFab_Api.js.ejs

```diff
@@ -151,11 +151,77 @@ if(!PlayFab._internalSettings) {
+                });
+            }
+            var resolvedPromise = await sendRequest(xhr);
+            return resolvedPromise;
         }
     }
 }
 
+if (typeof PlayFab._internalSettings.authenticationContext == "undefined") {
+    // Initialize the authenticationContext if it doesnt exist
+    // We could use a ternary operator here, but length of the line of code would effect readability
+    PlayFab._internalSettings.authenticationContext = {};
+}
+
+PlayFab._internalSettings.GetAuthValue = function (request, authKey) {
+    // Use the most-recently saved sessionTicket, unless one was provided in the request via the PlayFabId property
+    var authInfoMap = {
+        'X-EntityToken': 'entityToken',
+        'X-Authorization': 'sessionTicket',
+        'X-SecretKey': 'developerSecretKey'
+    };
+    var authAttr = authInfoMap[authKey];
+    var authValue = authAttr ==  'developerSecretKey' ? PlayFab.settings[authAttr]: PlayFab._internalSettings[authAttr];
+    if (request.hasOwnProperty('PlayFabId')) {
+        var playFabId = request.PlayFabId;
+        // Make sure that the authenticationContext exists before attempting to access it
+        if (PlayFab._internalSettings.authenticationContext.hasOwnProperty(playFabId)) {
+            if (PlayFab._internalSettings.authenticationContext[playFabId].hasOwnProperty(authAttr)) {
+                authValue = PlayFab._internalSettings.authenticationContext[playFabId][authAttr];
+            }
+        }
+    }
+    return authValue;
+};
+
+PlayFab._internalSettings.GetAuthInfo = function (request, authKey) {
+    // Use the most-recently saved sessionTicket, unless one was provided in the request via the PlayFabId property
+    var authInfoMap = {
+        'X-EntityToken': {
+            'authAttr': 'entityToken',
+            'authError': 'errorEntityToken'
+        },
+        'X-Authorization': {
+            'authAttr': 'sessionTicket',
+            'authError': 'errorLoggedIn'
+        },
+        'X-SecretKey': {
+            'authAttr': 'developerSecretKey',
+            'authError': 'errorSecretKey'
+        }
+    };
+    var authAttr = authInfoMap[authKey]['authAttr'];
+    var authValue = authAttr ==  'developerSecretKey' ? PlayFab.settings[authAttr]: PlayFab._internalSettings[authAttr];
+    if (request.hasOwnProperty('PlayFabId')) {
+        var playFabId = request.PlayFabId;
+        // Make sure that the authenticationContext exists before attempting to access it
+        if (PlayFab._internalSettings.authenticationContext.hasOwnProperty(playFabId)) {
+            if (PlayFab._internalSettings.authenticationContext[playFabId].hasOwnProperty(authAttr)) {
+                authValue = PlayFab._internalSettings.authenticationContext[playFabId][authAttr];
+            }
+        }
+    }
+    authAttr = authInfoMap[authKey]['authError'];
+    var authError = PlayFab._internalSettings[authAttr];
+
+    return {'authValue': authValue, 'authError': authError};
+};
```

Comment:
I feel like you're replacing 3 lines of admittedly awkward template code with 50 lines of "clever" code. I don't see it as an improvement, and it doesn't structurally match the way we're doing it in the other SDKs.